slay/truss_adapter/deploy.py

```diff
@@ -0,0 +1,325 @@
+    # TODO This assume all imports are absolute w.r.t workflow root (or site-packages).
+    # Also: apparently packages need an `__init__`, or crash.
+    _copy_python_source_files(workflow_root, pkg_dir / pkg_dir)
+
+    # TODO Truss package contains this from `{ include = "slay", from = "." }`
+    # pyproject.toml. But for quick dev loop just copy from local.
+    shutil.copytree(
+        os.path.dirname(slay.__file__),
+        pkg_dir / "slay",
+        dirs_exist_ok=True,
+    )
+    return truss_dir
+
+
+def get_api_key_from_truss_config() -> str:
+    return remote_factory.load_config().get("baseten", "api_key")
+
+
+class _BasetenEnv(enum.Enum):
+    LOCAL = enum.auto()
+    STAGING = enum.auto()
+    PROD = enum.auto()
+    DEV = enum.auto()
+
+
+def _infer_env(baseten_url: str) -> _BasetenEnv:
+    if baseten_url in {"localhost", "127.0.0.1", "0.0.0.0"}:
+        return _BasetenEnv.LOCAL
+
+    if "staging" in baseten_url:
+        return _BasetenEnv.STAGING
+
+    if "dev" in baseten_url:
+        return _BasetenEnv.DEV
+
+    return _BasetenEnv.PROD
+
+
+def _model_url(baseten_env: _BasetenEnv, model_id: str) -> str:
+    if baseten_env == _BasetenEnv.LOCAL:
+        return f"http://localhost:8000/models/{model_id}"
+
+    if baseten_env == _BasetenEnv.STAGING:
+        return f"https://app.staging.baseten.co/models/{model_id}"
+
+    if baseten_env == _BasetenEnv.DEV:
+        return f"https://app.dev.baseten.co/models/{model_id}"
+
+    return f"https://model-{model_id}.api.baseten.co/production"
+
+
+class BasetenClient:
+    """Helper to deploy models on baseten and inquire their status."""
+
+    # TODO: use rest APIs where possible in stead of graphql_query.
+    def __init__(self, baseten_url: str, baseten_api_key: str) -> None:
+        self._baseten_url = baseten_url
+        self._baseten_env = _infer_env(baseten_url)
+        self._baseten_api_key = baseten_api_key
+        self._remote_provider: truss_remote.TrussRemote = self._create_remote_provider()
+
+    def deploy_truss(self, truss_root: Path) -> definitions.BasetenRemoteDescriptor:
+        tr = truss.load(str(truss_root))
+        model_name = tr.spec.config.model_name
+        assert model_name is not None
+
+        logging.info(f"Deploying model `{model_name}`.")
+        service = self._remote_provider.push(
+            tr, model_name=model_name, trusted=True, publish=False
+        )
+        if service is None:
+            raise ValueError()
+        service = cast(b10_service.BasetenService, service)
+
+        model_service = definitions.BasetenRemoteDescriptor(
+            b10_model_id=service.model_id,
+            b10_model_version_id=service.model_version_id,
+            b10_model_name=model_name,
+            b10_model_url=_model_url(self._baseten_env, service.model_id),
+        )
+        return model_service
+
+    def get_model(self, model_name: str) -> definitions.BasetenRemoteDescriptor:
+        query_string = f"""
```

Comment:
@squidarth is there a better REST way of doing this now?

Reply:
unfortunately for this one, we don't have a way of getting a model by name in the REST API.

Reply:
Yes, already noted as TODO ;)